pkg/metricscollector/prommetrics.go

```diff
@@ -53,16 +54,25 @@ var (
 			Namespace: DefaultPromMetricsNamespace,
 			Subsystem: "scaler",
 			Name:      "metrics_value",
-			Help:      "Metric Value used for HPA",
+			Help:      "The current value for each scaler's metric that would be used by the HPA in computing the target average.",
 		},
 		metricLabels,
 	)
-	scalerMetricsLatency = prometheus.NewGaugeVec(
+	scalerMetricsLatencyDeprecated = prometheus.NewGaugeVec(
 		prometheus.GaugeOpts{
 			Namespace: DefaultPromMetricsNamespace,
 			Subsystem: "scaler",
 			Name:      "metrics_latency",
-			Help:      "Scaler Metrics Latency",
+			Help:      "DEPRECATED - will be removed in 2.15  use 'scaler_metrics_latency_seconds' instead.",
+		},
+		metricLabels,
+	)
+	scalerMetricsLatency = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: DefaultPromMetricsNamespace,
+			Subsystem: "scaler",
+			Name:      "metrics_latency_seconds",
```

Comment:
The unit is miliseconds, do we have to change the name from `metrics_latency_seconds` to `metrics_latency_miliseconds`?

Reply:
it is preferred in prometheus to actually use the "base" unit (seconds, watts, etc): https://prometheus.io/docs/practices/naming/#base-units I'd suggest to change to the base unit, but this isn't my project... ;)

Reply:
I'm not against changing the scale (using another metric, ofc), but I'm afraid about missing information. Can we guarantee that we won't miss info? Totally ignorant question from my side but it's just for double check if Prometheus handles float numbers properly (I don't have any Prometheus server to check it right now)

Reply:
To elaborate further, it might actually be better to add to the Otel metrics the `WithUnit` (https://pkg.go.dev/go.opentelemetry.io/otel/metric#WithUnit) call and specify the unit. There could be both `s` or `ms` defined (see https://ucum.org/ucum#section-Tables-of-Terminal-Symbols), and supply the value accordingly.

Reply:
Sorry for the spamming with comments, but dug a bit deeper in the otel docs, and on https://opentelemetry.io/docs/specs/semconv/general/metrics/#instrument-units, they state (last item in the list): 

> When instruments are measuring durations, seconds (i.e. s) SHOULD be used.

Reply:
Also, prometheus will handle seconds as a float64 for the value perfectly fine. Grafana or other solutions, when the unit is configured correctly, will show this correctly as milliseconds in the visualisations if the values are in that range.

Reply:
Luckily, we aren't using the unit for otel metrics, so we can just add it, can't? Could you include it as part of this PR? We could increase a bit the scope from prometheus metrics to metrics :)

Reply:
Reworked the Otel metrics:
- reworded the `totals` to `count` as preferred by Otel, and deprecated the old ones.
- added the "s" unit to the latency metrics
- added a test for the unit being recorded, and 500 milliseconds should show as 0.5

Two things I noticed: 

First, a bit of a meta-discussion, and we all know that naming things is hard in IT. I'm struggling a bit with the wording of `keda.triggers.count` (and in prom that's `keda_triggers_handled_total`). I'm kind of inclined towards renaming those to `keda.triggers.count` and `keda_triggers_total`, as it is the number of triggers, not the number of "handles" we're counting. Didn't touch the prom side of things for now, so it kind of diverges now. Feel free to suggest to either include `handled` or not, and I can align prom and otel.

Second, to my surprise, the e2e tests for Otel are actually ... prometheus tests (check https://github.com/kedacore/keda/blob/main/tests/sequential/opentelemetry_metrics/opentelemetry_metrics_test.go). I'm not going to rework that whole thing to Otel, but would it not be better to be "honest" and just wipe the tests, instead of testing something completely different and getting a false coverage number?

Reply:
For clarity, it used to be "keda_triggers_total", but I renamed it in prom to "keda_triggers_handled_total". However, looking at the code, it's an up-down counter, so it's not the historical ever-increasing number of handled triggers. It's the amount of triggers which are currently viewed/defined/loaded/registered/observed... from the CRDs.

So, the "handled" is a bad choice, my mistake. Just naming it "keda_triggers_total" is ambiguous (imho), as it is unclear what exactly about the trigger is being counted (invocations, definitions, registrations...). Exactly the same thoughts about "keda_resources_total" btw.

As this whole PR is biased by myself already, I suggest to switch to the naming of `triggers_registered_total` and `resources_registered_total` for prometheus, and to `triggers_registered_count` and `resources_registered_count` in Otel. It clearly indicated the fact that the system read, validated and started to use the trigger/CRD. I'll prep the PR in that direction.

Reply:
@zroubalik @tomkerkhove WDYT?

Reply:
kindly reminder @zroubalik  @tomkerkhove

Reply:
kindly reminder @zroubalik @tomkerkhove